scripts/requirements-toolchain.txt

```diff
@@ -44,33 +40,27 @@ lxml==4.6.2
 MarkupSafe==1.1.1
 mccabe==0.6.1
 milksnake==0.1.5
-more-itertools==8.2.0
 mock==4.0.3
 mypy==0.800
 mypy-extensions==0.4.3
-nanopb==0.4.4
 naturalsort==1.5.1
 packaging==20.8
 Pillow==8.1.0
 pluggy==0.13.1
 ply==3.11
 prettytable==2.0.0
 progress==1.5
-pyasn1==0.4.8
 psutil==5.8.0
 py==1.10.0
 pycparser==2.20
 pyelftools==0.27
-PyGithub==1.54
```

Comment:
Pygithub is still part of `requirements-extra.txt`, so should be kept so the toolchain is also useful for ourselves.

Reply:
extra is not included in the docker image.  are you sure you want it in the customer toolchain?

Reply:
well, if it is not included, then this test will fail:
https://github.com/nrfconnect/sdk-nrf/tree/master/tests/modules/lib/cddl-gen

as it relies on https://github.com/NordicSemiconductor/cddl-gen/blob/0680da699c5bddb97c287d71db90739ef2c7ccfe/scripts/cddl_gen.py#L8
which again relies on `regex`, which is defined here:
https://github.com/nrfconnect/sdk-nrf/blob/d375d0a68016b1b4a57e2ef88f1984b258423be0/scripts/requirements-extra.txt#L4

Users are not expected to run cddl-gen, but we still have test code, and I think it's important that even users are able to always run all tests in `sdk-nrf`, this I think that package is placed correctly in `extra`, but could of course be moved.

The test is also the reason it's part of requirments-ci.txt: 
https://github.com/nrfconnect/sdk-nrf/blob/d375d0a68016b1b4a57e2ef88f1984b258423be0/scripts/requirements-ci.txt#L82

@oyvindronningstad anything to add ?

Reply:
besides that, I still think it's valuable if the toolchain contains all packages that we also need for doing our job.
Otherwise, developers working both up-/downstream will still need to ensure they have those packages installed, and if they need to do so, they have less incentive to use the NCS Toolchain, and that is not something we want.
We really would like internal developers to use the exact same tools that are provided externally.

Reply:
I thought the purpose of requirements-extra.txt was that it was things most people don't use, but I guess its not a problem to add it.

Reply:
regex is in requirements-build.txt already.

Reply:
> I thought the purpose of requirements-extra.txt was that it was things most people don't use, but I guess its not a problem to add it.

Correct, so users doing a manual install are free to ignore the `requirements-extra.txt` if they don't need those things (which most people won't need).
But the toolchain should be useful to all users, including those that want to run a full twister or need those extra packages for special purposes. If we don't include extras in the toolchain, then such users would not be able to benefit from the toolchain.